### ISSUE TITLE
Orphans

### ISSUE THREAD

**Issue by jasongao97** [nature-of-code/noc-book-2#666]
> Orphans: We generally try to avoid words of four characters or fewer alone on the last line of a paragraph. I'm not sure if it's possible to fix these since we're not using standard typesetting software, but I have marked them with the following tag: #orphan.

<img width="719" alt="image" src="https://github.com/nature-of-code/noc-book-2023/assets/6762203/d5669a7c-3153-4df1-abdd-bf7d67701cb3">

> Holes: We generally try to fix holes that are at least six characters deep in the right margin.

<img width="724" alt="image" src="https://github.com/nature-of-code/noc-book-2023/assets/6762203/bd45191e-b5c2-4031-86de-712170121fcc">

I don't currently have a straightforward solution in mind. Perhaps we can add custom kerning adjustments for specific paragraphs?

**Comment by shiffman** [comment]
@jasongao97 is looking at this again with a possible solution.

**Comment by shiffman** [comment]
We will save #873 for another time, maybe document it as part of more general "magic book research", for now we'll implement a custom style callout to change the letter spacing for specfiic paragraphs if I want to try it:

<img width="806" alt="Screen Shot 2024-03-05 at 9 37 38 AM" src="https://github.com/nature-of-code/noc-book-2023/assets/191758/74938a67-acc6-4719-84d6-80b919eded45">

**Comment by shiffman** [comment]
Adding a note here that we'll need to ignore the line breaks for the web build: https://github.com/nature-of-code/noc-book-2023/pull/894#discussion_r1513252260

(Or i might just take them out of notion once the PDF is finished)

**Comment by shiffman** [comment]
The manual breaks is actually working quite well for orphans and holes as I can break a line before a hole to rebalance. I'm closing this issue and now we can open individual issues for specific orphans or holes if needed. I will file a new issue regarding ignoring `<br>` tags for the web build.
